src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -0,0 +1,227 @@
+        {
+            var query = new StringBuilder();
+            HttpClient client = new HttpClient
+            {
+                BaseAddress = new Uri("https://api.github.com")
+            };
+            client.DefaultRequestHeaders.Add("User-Agent", @"Mozilla/5.0 (Windows NT 10; Win64; x64; rv:60.0) Gecko/20100101 Firefox/60.0");
+
+            query.Append($"is:issue+is:open+{"Error Message: " + errorMessage}in:title&order=desc");
+            JObject responseContent;
+            using (HttpResponseMessage response = await client.GetAsync(
+                $"search/issues?q={query}"))
+            {
+                responseContent = JObject.Parse(await response.Content.ReadAsStringAsync());
+            }
+            JArray items = JArray.Parse(responseContent["items"].ToString());
+            if (!items.IsNullOrEmpty())
+            {
+                foreach (JObject item in items)
+                {
+                    var createdDate = item.GetValue("created_at");
+                    var result = createdDate.ToObject<DateTime>();
+                    if (result <= transactionDateTime)
+                    {
+                        Logger.LogInformation("The issue already exists so skipping this error");
+                        return false;
+                    }
+                    else
+                        return true;
+                }
+            }
+            return true;
+        }
+
+
+        private async Task CreateGitHubIssueAsync(DateTime transactionDateTime , string repositoryUrl, string branch, 
+            string action, String errorMessage, String method, string whereToCreateIssue)
+        {
+            Logger.LogInformation($"Something failed in the repository : {repositoryUrl} ");
+
+            string fyiHandles = "@epananth";
+            string gitHubToken = null, azureDevOpsToken = null;
+            string label = "UpdateDependency";
+
+            using (Logger.BeginScope($"Opening GitHub issue for RepositoryBranchUpdateHistory."))
+            {
+                try
+                {
+                    if (!string.IsNullOrEmpty(whereToCreateIssue))
+                    {
+                        IGitHubTokenProvider gitHubTokenProvider = Context.GetService<IGitHubTokenProvider>();
+                        long installationId = await Context.GetInstallationId(whereToCreateIssue);
+                        gitHubToken = await gitHubTokenProvider.GetTokenForInstallationAsync(installationId);
+
+                        Logger.LogInformation($"GitHub token acquired for '{whereToCreateIssue}'!");
+                    }
+
+                    IssueManager issueManager = new IssueManager(gitHubToken, azureDevOpsToken);
+
+                    string title = $"Repository Url :'{repositoryUrl}' Error :'{errorMessage}'";
+                    string description = $"Something failed during dependency update" +
```

Comment:
We really shouldn't use "Environment.NewLine" here, because the current machines newline has nothing to do with GitHub's newline.

That said, GitHub normalizes them, so it double doesn't matter, so this should just be a "literal string" in C#
```csharp
string description = @$"Failured during dependency update

RepositoryUrl: [{repositoryUrl}](repositoryUrl)

Branch Name : {branch}
```
and so on.